src/App.tsx

```diff
@@ -1,25 +1,47 @@
-import React from 'react';
-import logo from './logo.svg';
+import axios from 'axios';
+import { useEffect, useState } from 'react';
 import './App.css';
 
+export interface Breed{
+  name: string
+}
+export interface GetBreedsResponse {
+  data: Breed[]
+}
+
+export const getBreeds = async (): Promise<Breed[]|any>  => {
+  try{
+    return await axios.get('https://api.thecatapi.com/v1/breeds');
+  }catch(error){
+    return error;
+  }
+}
+
 function App() {
+  
+
+  
+  const [cats,setCats] = useState<Breed[]>([]);
+
+useEffect(()=>{ 
+  // bora chamar a API
+  getBreeds()
+    .then(({data}) => {
+      setCats(data)
+  })
```

Comment:
Imagino que você usou Promise aqui porque teve problemas chamando com `await`, porque não é possivel ter `async` no callback do hook.

Pra resolver isso a gente cria uma função:
```suggestion
useEffect(() = > { 
  const loadBreeds = async () => {
    const breeds = await getBreeds();
    setCats(breeds);
  }
  loadBreeds();
}, [setCats]);
```